doc/source/glossary.rst

```diff
@@ -2,10 +2,101 @@
 Glossary
 ********
 
-.. toctree::
-
 .. glossary::
 
+
+   (`n`,)
+       A tuple with one element. The trailing comma distinguishes a one-element
+       tuple from a parenthesized ``n``.
+
+
+   -1
+       Used as a dimension entry, ``-1`` instructs NumPy to choose the length
+       that will keep the total number of elements the same.
+
+
+   ``...``
+       An :py:data:`Ellipsis`
+
+       **When indexing an array**, shorthand that the missing axes, if they
+       exist, are full slices.
+
+           >>> a = np.arange(24).reshape(2,3,4)
+
+           >>> a[...].shape
+           (2, 3, 4)
+
+           >>> a[...,0].shape
+           (2, 3)
+
+           >>> a[0,...].shape
+           (3, 4)
+
+           >>> a[0,...,0].shape
+           (3,)
+
+       It can be used at most once; ``a[...,0,...]`` raises an ``IndexError``.
+
+       **In printouts**, NumPy substitutes ``...`` for the middle elements of
+       large arrays. To see the entire array, use
+       :doc:`numpy.printoptions. <reference/generated/numpy.printoptions>`
```

Comment:
Punctuation probably makes sense outside the link
```suggestion
       :doc:`numpy.printoptions <reference/generated/numpy.printoptions>`.
```
But is there any reason we can't just use
```suggestion
       `numpy.printoptions`
```
which should link to the function?